controller/src/beerocks/cli/beerocks_cli_main.cpp

```diff
@@ -215,6 +216,8 @@ static int cli_non_interactive(std::string path, std::string tmp_path, std::stri
     beerocks::cli *cli_ptr = &cli_soc;
     std::string bml_prefix(BML_PREFIX);
 
+    beerocks::string_utils::trim(command_string, "\"\'");
```

Comment:
It won't break quoted arguments because it only trims on the sides.
> This will break the handling of quoted arguments in parsedCommandArgumentsLinux, won't it?
> 
> In addition, I don't see why this would be needed... According to the commit message:
> 
> > The non-interactive mode was able to receive only command with no
> > arguments (such as 'bml_conn_map'),
>

Exactly, because you used quotes. If you  weren't, then optarg was equal to 'dump_node_info', without the rest of the command.
> Actually, I use commands with arguments in non-interactive mode all the time, e.g. `beerocks_cli -c "dump_node_info 00:11:22:33:00:20"`. It works for me...
> 
> > since the arguments string were
> > cutoff by getopt(). To force getopt() to take the argument one must
> > wrap the command and arguments with single or double-quotes.
> > But then it caused the command to not be found since it has quotes
> > note on its beginning.
> 
AFAIR, when I tested it there were quote characters on optarg.
> Err, no it doesn't... The quotes are interpreted by the shell, and the shell passes the unquoted argument (containing a space) in argv.
> 
> > ```
> > The solution is to trim these quotes on non-interactive mode.
> > ```
> 
> In other words, I don't see how/why this is needed.

Reply:
I will remove this.